R-package/tests/testthat/test_basic.R

```diff
@@ -2928,6 +2949,36 @@ test_that("lightgbm() defaults to 'regression' objective if objective not otherw
   expect_false(any(model_txt_lines == "objective=regression_l1"))
 })
 
+test_that("lightgbm() accepts 'num_threads' as either top-level argument or under params", {
+  bst <- lightgbm(
+    data = train$data
+    , label = train$label
+    , nrounds = 5L
+    , verbose = VERBOSITY
+    , num_threads = 1L
+  )
+  expect_equal(bst$params$num_threads, 1L)
+
+  bst <- lightgbm(
+    data = train$data
+    , label = train$label
+    , nrounds = 5L
+    , verbose = VERBOSITY
+    , params = list(num_threads = 1L)
+  )
+  expect_equal(bst$params$num_threads, 1L)
```

Comment:
I believe this check is quite "weak". It will be a lot better to inspect dumped LightGBM model to ensure that `num_threads` parameter reaches cpp side and is properly treated.

Here is an example in Python:
```
import numpy as np
import lightgbm as lgb

X = np.random.random((50, 10))
y = np.random.random(50)

est = lgb.LGBMRegressor(n_estimators=5, n_jobs=7).fit(X, y)
est.booster_.save_model('model.txt')
```
You can find `[num_threads: 7]` line in the `model.txt` file under the `parameters:` section.